tool/src/main.cpp

```diff
@@ -20,40 +20,183 @@
 
 #ifdef WITH_OPENCV
 #include <opencv2/opencv.hpp>
+#include <sys/ioctl.h>
+#include <linux/videodev2.h>
+#include <fcntl.h>
+#include <unistd.h>
 
 using namespace cv;
 #endif
 
 namespace fcwt {
 
+sock sockfd;
+
 log_settings log_conf;
 
+current_properties settings;
+std::timed_mutex g_comm_lock;
+
+// On X-T100 at least the auto-focus points are specified with these ranges.
+// Not sure how we get the ranges from the camera..
+//
+// The area around misses a selection of each of those points
+//
+// Needs to be in AF-S mode not manual
+#define POINTS_X 0xd
+#define POINTS_Y 0x7
+
+auto_focus_point requested_focus_point = 0;
+bool set_focus(int x, int y) {
+    if( x < 1 || x > POINTS_X || y < 1 || y > POINTS_Y )
+        return false;
+
+    // TODO: Refuse if not in AF-S or recording?
+
+    requested_focus_point.x = x;
+    requested_focus_point.y = y;
+
+    log(LOG_DEBUG, string_format("Set focus point %d x %d", x, y));
+
+    if (update_setting(sockfd, requested_focus_point)) {
+        // TODO: Decode if it got focused or not successfully (red/green bracket)
+        if (current_settings(sockfd, settings))
+          print(settings);
+    } else {
+        log(LOG_ERROR, string_format("Failed to adjust focus point"));
+        return false;
+    }
+    return true;
+}
+
 #ifdef WITH_OPENCV
-void image_stream_cv_main(std::atomic<bool>& flag) {
+#define WIN_NAME "Display Window"
+
+static void onMouse( int event, int x, int y, int, void* )
+{
+    if( event != EVENT_LBUTTONDOWN )
+        return;
+
+    Rect win_size = getWindowImageRect(WIN_NAME);
+    float x_perc = (float)x / win_size.width;
+    float y_perc = (float)y / win_size.height;
+
+    // We are running on display thread
+    if(g_comm_lock.try_lock_for(std::chrono::milliseconds(10))) {
+        set_focus(x_perc * (2+POINTS_X), y_perc * (2+POINTS_Y));
+        g_comm_lock.unlock();
+    } else {
+        log(LOG_ERROR, string_format("Failed to get control mutex"));
+    }
+}
+
+//#define CV_TEST
+
+int setup_v4l2(std::string v4l2lo_dev, Mat image) {
+    int v4l2lo = open(v4l2lo_dev.c_str(), O_WRONLY);
+    if(v4l2lo < 0) {
+        log(LOG_ERROR, string_format("Error opening v4l2l device: %s", strerror(errno)));
+        return v4l2lo;
+    }
+
+    struct v4l2_format v;
+    int t;
+    v.type = V4L2_BUF_TYPE_VIDEO_OUTPUT;
+    t = ioctl(v4l2lo, VIDIOC_G_FMT, &v);
+    if( t < 0 ) {
+        log(LOG_ERROR, string_format("ioctl error with setting up v4l2l device: %d", t));
+        close(v4l2lo);
+        return -1;
+    }
+
+    v.fmt.pix.width = image.cols;
+    v.fmt.pix.height = image.rows;
+    v.fmt.pix.pixelformat = V4L2_PIX_FMT_RGB24;
```

Comment:
```suggestion
    v.fmt.pix.pixelformat = V4L2_PIX_FMT_BGR24;
```

I have found a simple fix, by just changing the V4L2 pixel format to BGR, so it matches OpenCV's default BGR format.